test/benchmark_csv_reporter.cc

```diff
@@ -0,0 +1,87 @@
+#include <algorithm>
+#include <regex>
+
+#include <catch2/catch_test_case_info.hpp>
+#include <catch2/reporters/catch_reporter_registrars.hpp>
+#include <catch2/reporters/catch_reporter_streaming_base.hpp>
+#include <spdlog/fmt/fmt.h>
+#include <spdlog/fmt/ostr.h>
+
+// Escape according to RFC 4180 (w/o line break support)
+static std::string escape_csv(const std::string& str) {
+	assert(str.find_first_of("\r\n") == std::string::npos);
+	// Determine if we need escaping at all (string contains comma or quote).
+	if(str.find_first_of("\",") == std::string::npos) return str;
+
+	const std::regex re("\"");
+	// Prefix quotes with a another quote, wrap whole string in quotes.
+	return fmt::format("\"{}\"", std::regex_replace(str, re, "\"\""));
+}
+
+/**
+ * Prints benchmark results in CSV format.
+ * All timings are in nanoseconds.
+ *
+ * Note that unlike in some other reporters, sections that precede a BENCHMARK and are
+ * active in the current invocation will be printed as well (not just those directly
+ * surrounding a BENCHMARK).
+ */
+class benchmark_csv_reporter : public Catch::StreamingReporterBase {
+  public:
+	using StreamingReporterBase::StreamingReporterBase;
+
+	static std::string getDescription() { return "Reporter for benchmarks in CSV format"; }
+
+	void testRunStarting(Catch::TestRunInfo const& testRunInfo) override {
```

Comment:
We are west const everywhere else, so we should try to be consistent here (or change our policy, which I'm not opposed to either).

Reply:
I'm never sure how to deal with vendor interfaces that use different conventions. In this case I've opted to stick with Catch2's style for the function signatures (you'll notice the parameters are camelCased as well). But we can also mix it with our style. I don't have a strong preference, either feels weird to me.

Regarding our policy: The SYCL spec is west const, which sets a rather strong precedence I'd say.